workflows/iridanext.nf

```diff
@@ -64,14 +67,32 @@ workflow IRIDANEXT {
                fastq_2 ? tuple(meta, [ file(fastq_1), file(fastq_2) ]) :
                tuple(meta, [ file(fastq_1) ])}
 
-    SAMPLE_METADATA (
+    ASSEMBLY_STUB (
         input
     )
-    ch_versions = ch_versions.mix(SAMPLE_METADATA.out.versions)
+    ch_versions = ch_versions.mix(ASSEMBLY_STUB.out.versions)
+
+    // A channel of tuples of ({meta}, [read[0], read[1]], assembly)
+    ch_tuple_read_assembly = input.join(ASSEMBLY_STUB.out.assembly)
+
+    GENERATE_SAMPLE_JSON (
+        ch_tuple_read_assembly
+    )
+    ch_versions = ch_versions.mix(GENERATE_SAMPLE_JSON.out.versions)
+
+    GENERATE_SUMMARY (
+        ch_tuple_read_assembly.collect{ [it] }
+    )
+    ch_versions = ch_versions.mix(GENERATE_SUMMARY.out.versions)
+
+    SIMPLIFY_IRIDA_JSON (
+        GENERATE_SAMPLE_JSON.out.json
+    )
+    ch_versions = ch_versions.mix(SIMPLIFY_IRIDA_JSON.out.versions)
+    ch_simplified_jsons = SIMPLIFY_IRIDA_JSON.out.simple_json.map { meta, data -> data }.collect() // Collect JSONs
 
-    sample_metadata_ch = SAMPLE_METADATA.out.json.map { meta, data -> data }.collect()
     IRIDA_NEXT_OUTPUT (
```

Comment:
I'm wondering if you could also rename and adjust the location of the `output.json.gz` file. That is, the file should be renamed to `iridanext.output.json.gz` and it should be located in the output directory.

The location of the file I believe can be adjusted in a particular block in the `modules.config` file (by adjusting `publishDir`).

This is so that the output JSON file matches what is described here: https://github.com/phac-nml/pipeline-standards#32-irida-next-json

Reply:
My bad. I missed that. Updated.